clippy_lints/src/single_component_path_imports.rs

```diff
@@ -105,14 +119,22 @@ fn track_uses(
         ItemKind::Mod(_, ModKind::Loaded(ref items, ..)) => {
             check_mod(cx, &items);
         },
+        ItemKind::MacroDef(MacroDef { macro_rules: true, .. }) => {
```

Comment:
I had a brief look at [the macros 2.0 RFC](https://github.com/rust-lang/rust/issues/39412), and it seems like it will introduce a better way of declaring a macro's visibility:

```rust
pub(crate) macro foo { /* ... */ }
```

That's why I've included `macro_rules: true` so this only applies to `macro_rules!` macros, but I'm not too sure about this.